plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/ui/navigator/model/DatastoreIndexesDescriptor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.facets.ui.navigator.model;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.jface.viewers.StyledString;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+
+/**
+ * Represents a {@code datastore-indexes.xml}.
+ */
+public class DatastoreIndexesDescriptor extends AppEngineResourceElement {
+
+  public DatastoreIndexesDescriptor(IFacetedProject project, IFile datastoreIndexes) {
+    super(project, datastoreIndexes);
+  }
+
+  @Override
+  public StyledString getStyledLabel() {
+    return new StyledString("Datastore Indexes");
```

Comment:
Better, though I still wonder whether we should combine all these classes. They seem to differ only in a single string.

Reply:
Two reasons:
  1. Double-clicking the node or right-click > _Open_ opens the corresponding file.
  2. The nodes give a place to hang additional functionality (e.g., listing the cron entries, or the datastore indexes entity kinds).